File: javascript/easy/moveZeros.js
/**
 * @param {number[]} nums
 * @return {void} Do not return anything, modify nums in-place instead.
 */
var moveZeroes = function(nums) {
    let i = 0, j = 0;
    while(j < nums.length) {
        if(nums[j] !== 0) {
            nums[i++] = nums[j];
        }
        j++;
    }
    while(i < nums.length) {
        nums[i++] = 0;
    }
};

File: ruby/easy/move_zeros.rb
# @param {Integer[]} nums
# @return {Void} Do not return anything, modify nums in-place instead.
def move_zeroes(nums)
    i,j = 0,0
    while j < nums.length
        if(nums[j] != 0)
            nums[i] = nums[j]
            nums[j] = 0 if i != j
            i += 1
        end
        j += 1
    end
end
